packages/material-ui-lab/src/internal/pickers/PickersToolbarText.tsx

```diff
@@ -1,46 +1,73 @@
 import * as React from 'react';
 import clsx from 'clsx';
 import Typography, { TypographyProps } from '@material-ui/core/Typography';
-import { MuiStyles, StyleRules, WithStyles, withStyles } from '@material-ui/core/styles';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import { ExtendMui } from './typings/helpers';
+import pickersToolbarTextClasses, {
+  getPickersToolbarTextUtilityClass,
+} from './pickersToolbarTextClasses';
 
 export interface PickersToolbarTextProps extends ExtendMui<TypographyProps> {
+  component?: React.ElementType;
+  classes?: Partial<typeof pickersToolbarTextClasses>;
   selected?: boolean;
   value: React.ReactNode;
 }
 
-export type PickersToolbarTextClassKey = 'root' | 'selected';
-export const styles: MuiStyles<PickersToolbarTextClassKey> = (
-  theme,
-): StyleRules<PickersToolbarTextClassKey> => {
-  return {
-    root: {
-      transition: theme.transitions.create('color'),
-      color: theme.palette.text.secondary,
-      '&$selected': {
-        color: theme.palette.text.primary,
-      },
-    },
-    selected: {},
+export type PickersToolbarTextClassKey = keyof typeof pickersToolbarTextClasses;
+
+const useUtilityClasses = (styleProps: PickersToolbarTextProps) => {
+  const { selected, classes } = styleProps;
+
+  const slots = {
+    root: ['root', selected && 'selected'],
   };
-};
 
-const PickersToolbarText: React.FC<PickersToolbarTextProps & WithStyles<typeof styles>> = (
-  props,
-) => {
-  const { className, classes, selected, value, ...other } = props;
-
-  return (
-    <Typography
-      className={clsx(classes.root, className, {
-        [classes.selected]: selected,
-      })}
-      component="span"
-      {...other}
-    >
-      {value}
-    </Typography>
-  );
+  return composeClasses(slots, getPickersToolbarTextUtilityClass, classes);
 };
 
-export default withStyles(styles, { name: 'MuiPickersToolbarText' })(PickersToolbarText);
+const PickersToolbarTextRoot = experimentalStyled(
+  Typography,
+  {},
+  {
+    name: 'MuiPickersToolbarText',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)(({ theme }) => ({
+  transition: theme.transitions.create('color'),
+  color: theme.palette.text.secondary,
+  [`&.${pickersToolbarTextClasses.selected}`]: {
+    color: theme.palette.text.primary,
+  },
+}));
+
+const PickersToolbarText = React.forwardRef<HTMLSpanElement, PickersToolbarTextProps>(
+  function PickersToolbarText(inProps, ref) {
+    const props = useThemeProps({ props: inProps, name: 'MuiPickersToolbarText' });
+    const { component = 'span', className, selected, value, ...other } = props;
```

Comment:
```suggestion
    const { className, selected, value, ...other } = props;
```

As far as I can see from previous implementation, we should just propagate `span` there.

Reply:
without default value, the html will be h3.
![image](https://user-images.githubusercontent.com/18292247/116415856-03ddde00-a864-11eb-8537-43b2b93ffcb0.png)

Reply:
We should just propagate `component="span"` to the `Typography`, what I meant is we don't need the new component prop in the `PickersToolbarText`

Reply:
please review again